pytest_reportportal/service.py

```diff
@@ -102,8 +108,10 @@ def __init__(self):
         self.log_batch_size = 20
         self.log_item_id = None
         self.parent_item_id = None
+        self.project_settings = None
         self.rp = None
         self.rp_supports_parameters = True
+        self.test_threads_nested_steps = TestThreadsNestedSteps()
```

Comment:
I do not really understand why we use threads here. The step mechanism should be implemented as a separate pytest plugin. This new plugin can work with agent methods in order to report nested steps correctly. This work may require modification of the agent, however, this functionality is out of the agent's scope.